From a3e0f2b2b17efe3f08716e6ccd314af5a5c54510 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 29 Apr 2022 04:27:03 +0300
Subject: [PATCH] [SW-0] rom: tpm: store pcr in sha256 bank

As per security specification for gaudi3, PCR bank to be used
to extend the ppboot image sha is sha256. This is to align the
tpm implementation with gaudi2.

Change-Id: Iaab10a004b3f6ec5026601fae1a3bf67ae44cbfe
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 191af2c..af9698b 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -644,7 +644,7 @@
 	if (HL_FW_IMG_NOT_FOUND == idx)
 		return 1;
 
-	ret = tpm2_pcr_extend(TPM2_PPBOOT_PCR_IDX, TPM2_ALG_SHA384, desc.chksum, SHA384_SUM_LEN);
+	ret = tpm2_pcr_extend(TPM2_PPBOOT_PCR_IDX, TPM2_ALG_SHA256, desc.chksum, SHA384_SUM_LEN);
 	return ret;
 }
 
